public/app/plugins/datasource/tempo/traceql/situation.ts

```diff
@@ -105,21 +149,29 @@ export function getSituation(text: string, offset: number): Situation | null {
 
   const tree = parser.parse(text);
 
+  // Whitespaces (especially when multiple) on the left of the text cursor can trick the Lezer parser,
```

Comment:
interesting, is this something to add also to other autocomplete code like prometheus and loki?

Reply:
Is that relevant to the issue? I mean seems like the issue is just about white space before the cursor because of the lezer parser right? Both prometheus and loki use lezer parser.

Reply:
well it seems to affect the autocomplete there:

<img width="432" alt="Screenshot 2023-09-07 at 18 16 52" src="https://github.com/grafana/grafana/assets/1014802/16f6a121-2e43-4f88-af3b-5ee5e3442983">

So I still assume this fix would be valid there.